lib/widgets/event_dialog.dart

```diff
@@ -173,6 +157,36 @@ class _EventState extends State<EventDialog> {
   /// This widget creates an elipsis TextButton that only appears
   /// when the maximum title length has been exceeded. This will
   /// allow users to click and view the whole title should they desire.
+}
+
+class EventDialogTitle extends StatefulWidget {
+  const EventDialogTitle({super.key, required this.event});
+  final HDXEvent event;
```

Comment:
Why do you need the whole HDXEvent? I don't think any other fields of the event besides the title are used in this new Widget. Can you just pass a string of the title instead?